plugins/parquet/parquetembed.cpp

```diff
@@ -406,6 +410,70 @@ __int64 ParquetReader::next(TableColumns *&nextTable)
     return rowsProcessed++;
 }
 
+/**
+ * @brief Gets the information about the next row to be read from a Parquet file. A boolean is stored at the
+ * beginning of the memory buffer for the partition status. If the file is a partitioned dataset, the next row
+ * and the number of remaining rows are stored in the memory buffer. If the file is a regular Parquet file, the
+ * current and remaining tables are stored in the memory buffer. Additionally, the number of rows processed within
+ * the current table is stored in the memory buffer.
+ *
+ * @param cursor MemoryBuffer where file processing information is stored
+ * @return true If building the buffer succeeds
+ */
+bool ParquetReader::getCursor(MemoryBuffer & cursor)
+{
+    bool partition = endsWithIgnoreCase(partOption.c_str(), "partition");
+    cursor.append(partition);
+
+    // Adjust starting positions to current read position and remove
+    // already processed rows from the total count for the workers
+    if (partition)
+    {
+        cursor.append(startRow + totalRowsProcessed);
+        cursor.append(totalRowCount - totalRowsProcessed);
+    }
+    else
+    {
+        cursor.append(startRowGroup + tablesProcessed);
+        cursor.append(tableCount - tablesProcessed);
+        cursor.append(rowsProcessed);
+    }
+
+    return true;
+}
+
+/**
+ * @brief Resets the current access row in the Parquet file based on the information stored in a memory buffer
+ * created by getCursor. Sets restoredCursor to true for reading from the middle of the table in non-partitioned
+ * datasets. Resets the file read process trackers and reads the partition flag from the beginning of the buffer.
+ * If the file is a partitioned dataset then only the starting and remaining rows are read. Otherwise the
+ * starting and remaining tables are read as well as the current row within the table.
+ *
+ * @param cursor MemoryBuffer where file processing information is stored
+ */
+void ParquetReader::setCursor(MemoryBuffer & cursor)
+{
+    restoredCursor = true;
+    tablesProcessed = 0;
+    totalRowsProcessed = 0;
+    rowsProcessed = 0;
+    rowsCount = 0;
+
+    bool partition;
+    cursor.read(partition);
```

Comment:
You could sanity check that partition matches endsWithIgnoreCase(partOption.c_str(), "partition");

Reply:
I think that is a good check to have. Should I create a seperate JIRA for it, or wait for changes from another subtask?

I did test the setCursor/getCursor by initializing a new ParquetReader and stepping backwards through a random section of the table.